src/main/java/net/sf/jabref/logic/error/StreamEavesdropper.java

```diff
@@ -30,13 +34,35 @@ public static StreamEavesdropper eavesdropOnSystem() {
     }
 
     public TeeStream getOutStream() {
-        PrintStream consoleOut = new PrintStream(outByteStream);
-        return new TeeStream(consoleOut, systemOut, MessageType.OUTPUT);
+        PrintStream consoleOut = new PrintStream(outByteStream) {
+            @Override
+            public void write(byte[] buf, int off, int len) {
+                super.write(buf, off, len);
+                String s = new String(buf, off, len);
+                addToLog(s, MessageType.OUTPUT);
+            }
+        };
+        return new TeeStream(consoleOut, systemOut);
     }
 
     public TeeStream getErrStream() {
-        PrintStream consoleErr = new PrintStream(errByteStream);
-        return new TeeStream(consoleErr, systemErr, MessageType.EXCEPTION);
+        PrintStream consoleErr = new PrintStream(errByteStream){
+            @Override
+            public void write(byte[] buf, int off, int len) {
+                super.write(buf, off, len);
+                String s = new String(buf, off, len);
+                addToLog(s, MessageType.EXCEPTION
+                );
+            }
+        };
+        return new TeeStream(consoleErr, systemErr);
```

Comment:
If I understand it correctly, then everything from systemErr is also written to consoleErr. With your implementation all messages to consoleErr are posted to the cache (`addToLog`).
So far so good. But they also get posted as well as to the underlying `errByteStream`, which is no longer required (is it?). Thus could you remove the byte stream part?